R/dea-utils.R

```diff
@@ -0,0 +1,109 @@
+#' Create a matrix for input or output variables
+#'
+#' Create a matrix for input or output variables that can be used in DEA models
+#' from a supplied data.frame
+#'
+#' @param df A data.frame
+#' @param columns A vector of column names that should be included in the matrix
+#' @param id The name of the column with the DMU IDs
+#' @param normalize If `TRUE`, all columns will be normalized with a mean of 1
+#'
+#' @export
+create_matrix <- function(df, columns, id, normalize = FALSE) {
+  if (!inherits(df, 'data.frame')) {
+    cli::cli_abort('You must provide a valid data.frame')
+  }
+  if (!all(columns %in% colnames(df))) {
+    cli::cli_abort('All column names must exist in the supplied data.frame')
+  }
+  if (!id %in% colnames(df)) {
+    cli::cli_abort('ID column not found in the supplied data.frame')
+  }
+  x <- as.matrix(df[, columns])
+  rownames(x) <- as.vector(df[, id])
+  colnames(x) <- columns
+  x <- if (normalize) apply(x, 2, FUN = \(x) { x / mean(x) }) else x
+  x
+}
+
+#' Calculate scale efficiency
+#'
+#' Calculate scale efficiency from a set of inputs and outputs and return a
+#' data.frame
+#'
+#' @param inputs A matrix of inputs, created with `create_matrix`
+#' @param outputs A matrix of outputs, created with `create_matrix`
+#' @param orientation `in` for input oriented models or `out` for output oriented
+#' @param digits An integer with the number of digits to round to. If `NULL` the
+#'   values are kept unrounded.
+#'
+#' @export
+calculate_scale_eff <- function(
+    inputs,
+    outputs,
+    orientation = c('in', 'out'),
+    digits = NULL)
```

Comment:
We should start consolidating naming conventions in pioneeR with the parallel development in pear.  

* I have used `compute_*` instead of `calculate_*`, mainly because _compute_ is shorter. 
* I also use the full name for efficiency, eg. `compute_efficiency()`. 
* I use `x` and `y` for input/output, similar to `{Benchmarking}`. I'm not a big fan of abstract names, but I think there is an argument for x and y in our case, because they follow the convention in the literature. If not I suggest using the using _input_ / _output_  (wo/ the plural s).